snuba/query/matchers.py

```diff
@@ -201,6 +202,18 @@ def match(self, node: AnyType) -> Optional[MatchResult]:
         return MatchResult() if node == self.value else None
 
 
+@dataclass(frozen=True)
+class Datetime(Pattern[datetime]):
```

Comment:
I thought we had already built this

Reply:
I don't see any utility for matcing a specific datetime outside of tests. Normally we would match on the `datetime` type, and then in the test we could pull that value out and assert the specific value is correct, e.g.

`Literal(Param("timestamp", Any(datetime)))`

Reply:
ok, removed